examples/transactions/examples/with_access_list.rs

```diff
@@ -0,0 +1,60 @@
+//! Example of sending a EIP-1559 transaction with access list.
+use alloy::{
+    node_bindings::Anvil,
+    providers::{Provider, ProviderBuilder},
+    rpc::types::eth::{BlockId, TransactionRequest},
+    sol,
+};
+use eyre::Result;
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    SimpleStorage,
+    "examples/artifacts/SimpleStorage.json"
+);
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Spin up anvil node.
+
+    let anvil = Anvil::new().try_spawn()?;
+
+    // Create a provider.
+    let provider =
+        ProviderBuilder::new().with_recommended_fillers().on_builtin(&anvil.endpoint()).await?;
+
+    let alice = anvil.addresses()[0];
+    let bob = anvil.addresses()[1];
```

Comment:
new `default_signer` api with `on_anvil` helper is quite nice to make this easier / less verbose